Source/EmbeddedBoundary/WarpXFaceExtensions.cpp

```diff
@@ -21,6 +21,145 @@
 using namespace ablastr::fields;
 using warpx::fields::FieldType;
 
+namespace
+{
+
+    /**
+    * \brief Compute the minimal area for stability for the face i, j, k with normal 'dim'.
+    *
+    * \tparam dim normal direction to the plane in consideration (0 for x, 1 for y, 2 for z)
+    *
+    * \param[in] i, j, k the indices of the cell
+    * \param[in] lx, ly, lz arrays containing the edge lengths
+    * \param[in] dx, dy, dz the mesh with in each direction
+    */
+    template <int dim>
+    [[nodiscard]]
+    AMREX_GPU_DEVICE AMREX_FORCE_INLINE
+    amrex::Real
+    ComputeSStab(const int i, const int j, const int k,
+             const amrex::Array4<const amrex::Real> lx,
+             const amrex::Array4<const amrex::Real> ly,
+             const amrex::Array4<const amrex::Real> lz,
+             const amrex::Real dx, const amrex::Real dy, const amrex::Real dz)
+    {
+
+        using namespace amrex::literals;
+
+        if constexpr (dim == 0) {
+            return 0.5_rt * std::max({ly(i, j, k) * dz, ly(i, j, k + 1) * dz,
+                                  lz(i, j, k) * dy, lz(i, j + 1, k) * dy});
+        }
+        else if constexpr (dim == 1)
+        {
+#ifdef WARPX_DIM_XZ
+            return 0.5_rt * std::max({lx(i, j, k) * dz, lx(i, j + 1, k) * dz,
+                                  lz(i, j, k) * dx, lz(i + 1, j, k) * dx});
+#elif defined(WARPX_DIM_3D)
+            return 0.5_rt * std::max({lx(i, j, k) * dz, lx(i, j, k + 1) * dz,
+                                  lz(i, j, k) * dx, lz(i + 1, j, k) * dx});
+#else
+            amrex::Abort("ComputeSStab: Only implemented in 2D3V and 3D3V");
+#endif
+        }
+        else if constexpr(dim == 2){
+            return 0.5_rt * std::max({lx(i, j, k) * dy, lx(i, j + 1, k) * dy,
+                                  ly(i, j, k) * dx, ly(i + 1, j, k) * dx});
```

Comment:
Do we want to maintain the original alignment?
```suggestion
            return 0.5_rt * std::max({lx(i, j, k) * dy, lx(i, j + 1, k) * dy,
                                      ly(i, j, k) * dx, ly(i + 1, j, k) * dx});
```